libraries/adaptive-expressions/src/functionUtils.ts

```diff
@@ -779,4 +809,21 @@ export class FunctionUtils {
             return `${childExpr} in ${expr} is not any of [${typesStr}].`;
         }
     }
+
+    /**
+     * Helper function of get the number of properties of an object.
+     * @param obj An object.
+     */
+    private static getPropertyCount(obj: unknown): number {
+        let count = -1;
+        if (!Array.isArray(obj)) {
+            if (obj instanceof Map) {
+                count = obj.size;
+            } else if (typeof obj === 'object' && !(obj instanceof Date)) {
+                count = Object.keys(obj).length;
```

Comment:
`Object.keys(null)` and `Object.keys(undefined)` both throw - the else check should include `obj != null`